pkg/apis/camel/v1/maven_types_support.go

```diff
@@ -24,7 +24,11 @@ import (
 func (in *MavenArtifact) GetDependencyID() string {
 	switch {
 	case in.Version == "":
-		return "mvn:" + in.GroupID + ":" + in.ArtifactID
+		if in.Classifier == "" {
+			return "mvn:" + in.GroupID + ":" + in.ArtifactID
+		} else {
+			return "mvn:" + in.GroupID + ":" + in.ArtifactID + ":" + in.Type + ":" + in.Classifier
```

Comment:
Maybe we should check the presence of the type as well, is it?

Reply:
The definition of the full dependency is: `<group>:<artifact>:<type>:<version>:<classifier>`
However the dependency regex and the other checkers were not handling the `type` field, and I didn't want to spend too much time finding the right regex to handle the type or the classifier in isolation, because doing so would require to change the dependency spec to allow either `<group>:<artifact>:<type>` or `<group>:<artifact>:<classifier>`, but the `type`  and `classifier` may be any word, so we would have to determine a way to set both cases.
Then, the solution I found at this moment is if the `classifier` is set, then the `type` is a required field.
I suggest to open a new issue to better investigate this particular issue.

Reply:
I changed the behavior and added more testing.